haskell/runtime/src/ADL/Sys/Types.hs

```diff
@@ -61,5 +62,27 @@ type Maybe = Prelude.Maybe
 
 type Pair a b = (a,b)
 
+data Result t e
```

Comment:
This code should not be here. Since generateOrigADLType is `empty`,  it should be just

```
type Result a b = Prelude.Either a b
```

I think this is a bootstrapping issue, in the way that the compiler is written in haskell and bakes in some generated haskell. Can you try running this script:

https://github.com/timbod7/adl/blob/master/haskell/tools/scratch-build.sh

Reply:
I did run scratch-build. Nothing changed.

Reply:
OK. I remember now. You need to update `BootstrapCustomTypes.hs`

It's a "chicken and egg" situation where haskell custom types in the standard library need to be declared explicitly in haskell code as well in the annotations.

Reply:
Added.